### PR TITLE
New post: What are Monads?

### PR DESCRIPTION
**Pull request by adharshkamath**
A brief intro to monads in the context of FP

### CODE REVIEW

File: blog_src/_data/authors.yml [hunk 1]
```diff
@@ -710,7 +710,7 @@ abhinavpy:
   github: abhinavpy
   author_image: abhinavpy.jpg
   email: abhinavpy@ieee.org
-  user_description: "It\'s not whether you get knocked down, it\'s whether you get up."
+  user_description: It\'s not whether you get knocked down, it\'s whether you get up.
```
Comment (rohit-mp) [inline blog_src/_data/authors.yml]: Why was this changed?

Comment (adharshkamath) [reply blog_src/_data/authors.yml]: The " was giving an error. Worked fine after removing it.

> Why was this changed?

The " was causing an error. Worked fine after removing it.

Comment (adharshkamath) [reply blog_src/_data/authors.yml]: Everyone faced a similar issue it seems. Harsh put this solution on the group

Comment (nimbus98) [reply blog_src/_data/authors.yml]: @rohit-mp it seemed like that error was stopping the build with errors in many junior's local machines.

**Comment by adharshkamath** [comment]
### Screenshots:
Screenshot from 2019-10-04 16-38-46
Screenshot from 2019-10-04 16-38-52
Screenshot from 2019-10-04 16-38-56
Screenshot from 2019-10-04 16-39-01
Screenshot from 2019-10-04 16-39-07

**Comment by adharshkamath** [comment]
Screenshot from 2019-10-09 23-49-58
Screenshot from 2019-10-09 23-50-02
Screenshot from 2019-10-09 23-50-17
Screenshot from 2019-10-09 23-50-23
Screenshot from 2019-10-09 23-50-30

File: blog_src/_data/authors.yml (hunk 1 above)
Comment (salman-bhai) [reply blog_src/_data/authors.yml]: Apparently those are special characters that jekyll doesn't recognise. Usually happens when you copy paste from somewhere

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
+---
+layout: post
+title: "What are Monads?"
+author_github: adharshkamath
+date: 2019-10-3 23:00:00
```
Comment (salman-bhai) [inline blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Change day to 03 to maintain uniformity

File: blog_src/_data/authors.yml [hunk 3]
```diff
@@ -757,4 +757,9 @@ nimbus98:
   github: nimbus98
   author_image: 'akashnair.jpeg'
   user_description: 'I like Food'
-  email: akash30121998@gmail.com
\ No newline at end of file
+  email: akash30121998@gmail.com
+adharshkamath:
+  name: Adharsh Kamath
+  github: adharshkamath
+  author_image: blog-author.jpg
```
Comment (salman-bhai) [inline blog_src/_data/authors.yml]: Use quotes preferably. If there is a space in the image will cause issues

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+image: '/assets/img/'
+description: 'A brief Intro to Monads in FP'
+tags:
+- Functional Programming
+- Monads
+- Functional JavaScript
+categories:
+- Compsoc
```
Comment (salman-bhai) [inline blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: CompSoc

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+  
+
+Suppose you have two functions `CapitalizeFirstName(name)` and `CapitalizeLastName(name)`. The functions take in a name in the format `"firstname lastname"` and return `"Firstname lastname"` and `"firstname Lastname"` respectively (No other inputs are provided). Your task is to capitalize both the first and last names in a name.
+
+```js 
+var CapitalizeFirstName = function(name) {
+                     //do Some stuff
+                     return(FirstNameCapitalized);
+                     }
```
Comment (salman-bhai) [inline blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Alignment issues?

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+var CapitalizeLastName = function(name) {
+                     //do Some stuff
+                     return(LastNameCapitalized);
+                     }
```
Comment (salman-bhai) [inline blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Alignment issues?

**Comment by salman-bhai** [comment]
Please resolve pending conversations if you have addressed it !

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md (hunk 2 above)
Comment (adharshkamath) [reply blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Done!

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md (hunk 4 above)
Comment (adharshkamath) [reply blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Done!

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md (hunk 5 above)
Comment (adharshkamath) [reply blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Corrected!

File: blog_src/_posts/2019-10-3-monads-in-functional-programming.md (hunk 6 above)
Comment (adharshkamath) [reply blog_src/_posts/2019-10-3-monads-in-functional-programming.md]: Fixed!

File: blog_src/_data/authors.yml (hunk 3 above)
Comment (adharshkamath) [reply blog_src/_data/authors.yml]: Corrected!
